code/headers/gcode_generator.hpp

```diff
@@ -0,0 +1,163 @@
+#pragma once
+#include <hwlib.hpp> // size_t, (u)intx_t
```

Comment:
size_t and (u)intx_t can be included using cstdint and cstddef

The size_t is in the namespace std:: in cstddef.